docs/runtimes/create-and-use-functions.ipynb

```diff
@@ -0,0 +1,554 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "41aeed94",
+   "metadata": {},
+   "source": [
+    "![MLRun Function](../_static/images/mlrun_concepts_architecture.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "aa49c285",
+   "metadata": {},
+   "source": [
+    "An MLRun project object has a method called `set_function` which is a one-size-fits-all way of creating an MLRun function. The `set_function` method accepts a variety of sources including Python files, Jupyter notebooks, Git repos, and more.\n",
+    "\n",
+    "Additionally, the `set_function` method has a parameter called `with_repo` which denotes whether a function requires additional files or dependencies such as a Git repo or archive file. This Git repo or archive file is specified on a project level which the function will consume.\n",
+    "> Note: When using `with_repo`, the contents of the Git repo or archive file will be available in the current working directory of your MLRun function during runtime "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "14800073",
+   "metadata": {},
+   "source": [
+    "## Creating a Function"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e2311f14",
+   "metadata": {},
+   "source": [
+    "There are 3 main scenarios for creating a function:\n",
+    "1. [Single Source File](#single-source-file) - when your code can be contained into a single file\n",
+    "2. [Multiple Source Files](#multiple-source-files) - when your code requires additional files or dependencies\n",
+    "3. [Import Existing Function](#import-existing-function) - when your function already exists elsewhere and you just want to import it\n",
+    "\n",
+    "Using the `set_function` method of an MLRun project allows for each of these scenarios in a transparent way. Depending on the source passed in, the project will register the function using some lower level functions.\n",
+    "> Note: For specific use cases, you also have access to these lower level functions. These will be discussed in more detail at a later point\n",
+    "\n",
+    "When creating a new function, consider the following flow:"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "692a2bd9",
+   "metadata": {},
+   "source": [
+    "![Create function flow](../_static/images/mlrun_function_create_flow.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "70030b5f",
+   "metadata": {},
+   "source": [
+    "<a id=\"single-source-file\"></a>\n",
+    "### Single Source File (Python/Notebook)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1c137d9a",
+   "metadata": {},
+   "source": [
+    "The simplest way to create a function is to use a single file as the source. The code itself will be embeded into the MLRun function object. This makes the function quite portable as it does not depend on any external files. You can use any source file supported by MLRun such as Python or Jupyter notebook.\n",
```

Comment:
"The simplest way to create a function is to use a single file as the source. The code itself is embedded into the MLRun function object. This makes the function quite portable as it does not depend on any external files. You can use any source file supported by MLRun such as Python or Jupyter notebook.\n",
    "\n",
    "> Note: The return value of `set_function` is your MLRun function. You can immediately run it or apply additional configuration like resources, scaling, etc."